plasmapy/classes/simulation.py

```diff
@@ -0,0 +1,419 @@
+"""
+plasmapy.classes.simulation
+============================
+
+Classes and functionality for simulations.
+"""
+
+import numpy as np
+import astropy.units as u
+from .spatial_solvers import Solver
+from ..constants import mu0
+
+
+class MHDSimulation():
+    """Physics class for magnetohydrodynamics.
+
+    This class defines the MHD equations and implements time-stepping them:
+
+    .. math::
+
+       \\frac{\\partial \\rho}{\\partial t} + \\nabla \\cdot (\\vec{v} \\rho) = 0
+       \\frac{\\partial (\\rho \vec{v})}{\\partial t} + \\nabla \\cdot (\\vec{v} \\rho \\vec{v}) + \\nabla p = 0
+       \\frac{\\partial e}{\\partial t} + \\nabla \\cdot (\\vec{v} e + \\vec{v} p) = 0
+
+    for the fluid density, :math:`\\rho`, momentum, :math:`\\vec{m} = \\vec{v} \\rho`, energy :math:`e` and kinetic pressure :math:`p`.
+    The pressure is a derived quantity defined as
+
+    .. math::
+       p = (\\gamma - 1) (e - \\frac{\\rho \\vec{v}^2}{2})
+
+    Parameters
+    ----------
+    plasma : plasmapy.Plasma
+        Plasma object describing variables that are being solved for, etc.
+    gamma : float
+        Value of the adiabatic index.
+
+    Attributes
+    ----------
+    grid_size : tuple of ints
+        Size of the simulation grid, as defined at initiation.
+    gamma : float
+        Adiabatic index for the simulation.
+    """
+    def __init__(self, plasma):
+        """
+        """
+        self.dt = 0.01 * u.s
+        self.current_iteration = 0
+        self.current_time = 0 * u.s
+        self.plasma = plasma
+        # Domain size
+        # self.grid_size = grid_size
+
+        # Physical parameters
+        # self.gamma = gamma
+
+        grids = (self.plasma.x.si, self.plasma.y.si, self.plasma.z.si)
+        ranges = [grid for grid in grids if len(grid) > 1]
+        stepsize = [range[1] - range[0] for range in ranges] * grids[0].unit
```

Comment:
Good catch.  I am pretty sure I am doing things like this elsewhere, which I've been needing to go back and fix.

[For my own future reference in case I forget (hi future Nick!), I'm thinking of atomic.py with charge_state being a function in some places and an integer elsewhere]